core/src/saros/session/SarosCoreSessionContextFactory.java

```diff
@@ -48,11 +45,10 @@ public final void createComponents(ISarosSession session, MutablePicoContainer c
     else container.addComponent(ClientSessionTimeoutHandler.class);
 
     // Watchdogs
-    // FIXME this should only be added to the host context
-    container.addComponent(ConsistencyWatchdogHandler.class);
-
-    if (session.isHost()) container.addComponent(ConsistencyWatchdogServer.class);
-    else container.addComponent(ConsistencyWatchdogClient.class);
+    if (session.isHost()) {
+      container.addComponent(ConsistencyWatchdogServer.class);
+      container.addComponent(ConsistencyWatchdogHandler.class);
+    } else container.addComponent(ConsistencyWatchdogClient.class);
```

Comment:
I would prefer to use the braces for the else case as well to improve readability.

```suggestion
    } else {
      container.addComponent(ConsistencyWatchdogClient.class);
    }
```